Weird static libraries in libzip pkgconfig

On my machine, the file contents are:

```
prefix=/mingw64
exec_prefix=${prefix}
bindir=${exec_prefix}/bin
libdir=${exec_prefix}/lib
includedir=${prefix}/include

zipcmp=${exec_prefix}/bin/zipcmp

Name: libzip
Description: library for handling zip archives
Version: 1.6.1
Libs:  -L${libdir} -lzip
Libs.private:  -lD:/msys64/mingw64/lib/libz.dll.a -lD:/msys64/mingw64/lib/libbz2.dll.a -lD:/msys64/mingw64/lib/liblzma.dll.a -lbcrypt -ladvapi32
Cflags: -I${includedir}
```

Note `-lD:/msys64/mingw64/lib/libz.dll.a`. I don't have a `D:` drive, so that path doesn't actually exist. Also, there's a `-l` in front of a static library. Anyway, it was breaking my build.

I was able to get my build working again by manually editing the pkgconfig file to have the following:

```
prefix=/mingw64
exec_prefix=${prefix}
bindir=${exec_prefix}/bin
libdir=${exec_prefix}/lib
includedir=${prefix}/include

zipcmp=${exec_prefix}/bin/zipcmp

Name: libzip
Description: library for handling zip archives
Version: 1.6.1
Libs:  -L${libdir} -lzip
Libs.private:  C:/msys64/mingw64/lib/libz.dll.a C:/msys64/mingw64/lib/libbz2.dll.a C:/msys64/mingw64/lib/liblzma.dll.a -lbcrypt -ladvapi32
Cflags: -I${includedir}
```

See also #6734.

My guess would be because the static build gets installed over the shared one here (the order needs to be reversed): https://github.com/msys2/MINGW-packages/blob/52c6513cf726d9b39c29008fe2d54c32b374e01a/mingw-w64-libzip/PKGBUILD#L62

Sadly libzip doesn't build here:

```
[100%] Built target fopen_unchanged
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles/ziptool.dir/objects.a(ziptool.c.obj):ziptool.c:(.text.startup+0x41b): undefined reference to `__imp__snprintf.constprop.0'
[100%] Built target in-memory
collect2.exe: error: ld returned 1 exit status
```

Pkgconfig file should be similar to one in Arch Linux so it works with both static and shared:
```
Libs.private:  -lz -lbz2 -llzma -lgnutls -lnettle
```

> Sadly libzip doesn't build here atm (undefined reference to '__imp__snprintf.constprop.0'):

It's GCC 10 regression (no idea if reported upstream). Libzip builds fine with Clang.

About the `undefined reference to __imp__snprintf.constprop.0` error: removing the line `#define snprintf _snprintf` from `lib/compat.h` seems to fix that.

--- a/mingw-w64-libzip/0001-libzip-pkgconfig-add-cflags-private.patch
+++ b/mingw-w64-libzip/0001-libzip-pkgconfig-add-cflags-private.patch
@@ -1,0 +1,7 @@
+--- a/libzip.pc.in
++++ b/libzip.pc.in
+@@ -12,3 +12,4 @@
+ Libs: @PKG_CONFIG_RPATH@ -L${libdir} -lzip
+ Libs.private: @LIBS@
+ Cflags: -I${includedir}
++Cflags.private: -DZIP_STATIC

--- a/mingw-w64-libzip/PKGBUILD
+++ b/mingw-w64-libzip/PKGBUILD
@@ -6,9 +6,10 @@ _realname=libzip
 pkgbase=mingw-w64-${_realname}
 pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}"
 pkgver=1.10.1
-pkgrel=1
+pkgrel=2
 pkgdesc="A C library for reading, creating, and modifying zip archives (mingw-w64)"
-url="https://libzip.org"
+url='https://libzip.org/'
+msys2_repository_url='https://github.com/nih-at/libzip/'
 msys2_references=(
   "cpe: cpe:/a:libzip:libzip"
   "cpe: cpe:/a:nih:libzip"
@@ -26,15 +27,22 @@ makedepends=("${MINGW_PACKAGE_PREFIX}-cmake"
              "${MINGW_PACKAGE_PREFIX}-ninja"
              "${MINGW_PACKAGE_PREFIX}-cc"
              "groff")
-source=("https://libzip.org/download/${_realname}-${pkgver}.tar.xz")
-sha512sums=('d24670ca3fa5def0345d6c531045683f11a124af7975dd724eae3bf3128a62fb8185a3d4df029c7c8232c7584e5ac55d18c431dc53b91e992fbbc8d119442930')
+source=("https://libzip.org/download/${_realname}-${pkgver}.tar.xz"
+        "0001-libzip-pkgconfig-add-cflags-private.patch")
+sha512sums=('d24670ca3fa5def0345d6c531045683f11a124af7975dd724eae3bf3128a62fb8185a3d4df029c7c8232c7584e5ac55d18c431dc53b91e992fbbc8d119442930'
+            '5426b438300441c5fd36d1497f9a8a453f773a5ebd4e0affe90dc3933d84ec1f0f168e27b2c3b62587927f88c0c5cae3fc7bc6aee00d98116098d585ceeb7716')
 
 prepare() {
   cd "${srcdir}/${_realname}-${pkgver}"
 
+  # https://github.com/nih-at/libzip/pull/434
+  patch -p1 -i "${srcdir}/0001-libzip-pkgconfig-add-cflags-private.patch"
 }
 
 build() {
+  # Workaround for gcc 14 compiler error
+  CFLAGS+=" -Wno-error=incompatible-pointer-types"
+
   mkdir -p ${srcdir}/build-${MSYSTEM}-shared && cd ${srcdir}/build-${MSYSTEM}-shared
 
   MSYS2_ARG_CONV_EXCL="-DCMAKE_INSTALL_PREFIX=" \

